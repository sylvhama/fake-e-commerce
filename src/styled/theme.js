const theme = {
  color: {
    primary: '#ea352d',
    appBackground: '#f5f5f5',
    white: '#fff',
    lightGrey: '#ddd',
    grey: '#333',
    darkGrey: '#212121'
  },
  font: { h1: '2.5rem', h2: '2rem', h3: '1.5rem', p: '1rem' },
  size: {
    categories: '40px',
    footer: '60px',
    container: {
      reg: '940px',
      lg: '1020px'
    }
  },
  spacing: {
    sm: '8px',
    reg: '16px',
    lg: '24px'
  },
  radius: '4px'
};

export default theme;
