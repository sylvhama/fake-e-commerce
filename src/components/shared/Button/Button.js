import styled from 'styled-components';
import PropTypes from 'prop-types';

import lightenDarkenColor from '../../../styled/lightenDarkenColor';

const Button = styled.button`
  border: 1px solid ${props => props.theme.color[props.color]};
  padding: ${props => `${props.theme.spacing.sm} ${props.theme.spacing.reg}`};
  line-height: 1;
  font-size: ${props => props.theme.font.p};
  font-weight: ${props => (props.bold ? 'bold' : 'normal')};
  color: ${props =>
    props.color === 'primary'
      ? props.theme.color.white
      : props.theme.color.grey};
  background-color: ${props => props.theme.color[props.color]};
  border-radius: ${props =>
    props.rounded ? props.theme.font.p : props.theme.radius};
  cursor: pointer;
  transition: background-color 0.25s ease;
  &:hover {
    background-color: ${props =>
      lightenDarkenColor(props.theme.color[props.color], 20)};
  }
`;

Button.propTypes = {
  color: PropTypes.string,
  rounded: PropTypes.bool,
  bold: PropTypes.bool
};

Button.defaultProps = {
  color: 'lightGrey'
};

export default Button;
