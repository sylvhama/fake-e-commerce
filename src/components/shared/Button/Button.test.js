import React from 'react';
import 'jest-styled-components';

import { renderWithTheme } from '../../../tests/helpers';
import theme from '../../../styled/theme';

import Button from './Button';

test('Button renders correctly', () => {
  const rendered = renderWithTheme(<Button>Button</Button>).toJSON();
  expect(rendered).toMatchSnapshot();
  expect(rendered).toHaveStyleRule('font-weight', 'normal');
  expect(rendered).toHaveStyleRule('background-color', theme.color.lightGrey);
  expect(rendered).toHaveStyleRule('border-radius', theme.radius);
});

test('Button renders correctly with props', () => {
  const rendered = renderWithTheme(
    <Button bold rounded color="primary">
      Button
    </Button>
  ).toJSON();
  expect(rendered).toMatchSnapshot();
  expect(rendered).toHaveStyleRule('font-weight', 'bold');
  expect(rendered).toHaveStyleRule('background-color', theme.color.primary);
  expect(rendered).toHaveStyleRule('border-radius', theme.font.p);
});
