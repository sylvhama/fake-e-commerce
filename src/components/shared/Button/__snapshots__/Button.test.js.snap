// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  border: 1px solid #ddd;
  padding: 8px 16px;
  line-height: 1;
  font-size: 1rem;
  font-weight: normal;
  color: #333;
  background-color: #ddd;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: background-color 0.25s ease;
  transition: background-color 0.25s ease;
}

.c0:hover {
  background-color: #f1f1f1;
}

<button
  className="c0"
  color="lightGrey"
>
  Button
</button>
`;

exports[`Button renders correctly with props 1`] = `
.c0 {
  border: 1px solid #ea352d;
  padding: 8px 16px;
  line-height: 1;
  font-size: 1rem;
  font-weight: bold;
  color: #fff;
  background-color: #ea352d;
  border-radius: 1rem;
  cursor: pointer;
  -webkit-transition: background-color 0.25s ease;
  transition: background-color 0.25s ease;
}

.c0:hover {
  background-color: #fe4941;
}

<button
  className="c0"
  color="primary"
>
  Button
</button>
`;
